workflow_overview.md

```diff
@@ -0,0 +1,57 @@
+
+# 📄 General Overview of Project Workflow
+
+Once your [onboarding](https://intranet.neuro.polymtl.ca/onboarding/README.html) is complete, you will be ready to tackle your project!
+
+## 🖥️ Setting up 🖥️
+
+**Step 1.**
+* Make sure that your VPN connection is established or that you are connected to the Polytechnique wifi.
+
+**Step 2.**
+* Log in to one of the available [Neuropoly compute nodes](https://intranet.neuro.polymtl.ca/computing-resources/neuropoly/README.html):
+```
+ssh <POLYGRAMES_USERNAME>@<STATION>.neuro.polymtl.ca
+```
+
+**Step 3.**
+* Create your project working directory:
+```
+cd data_nvme_<POLYGRAMES_USERNAME>
+mkdir <PROJECT_NAME>
+cd <PROJECT_NAME>
+```
+
+**Step 4. Developing version-controlled software**
+* Ideally, you are working on code in Github repository (either a branch of an existing repo, or a new one that you created).
+* After adding your NeuroPoly workstation [SSH key to your Github account](https://docs.github.com/en/authentication/connecting-to-github-with-ssh/adding-a-new-ssh-key-to-your-github-account?platform=linux), you are ready to make a local fork of that remote repository:
+```
+cd data_nvme_<POLYGRAMES_USERNAME>/<PROJECT_NAME>
+git clone -b "<YOUR_WORKING_BRANCH>" git@github.com:<REPOSITORY>.git
+```
+
+**Step 5. The data**
+* It is critical to make sure that you know what data you are working with. 
+* Ideally, it should be in [BIDS](https://bids-specification.readthedocs.io/en/stable/) format on the [`data.neuro`](https://intranet.neuro.polymtl.ca/data/git-datasets.html) storage node: `data.neuro:datasets/<PROJECT_DATASET>`.
+* Thanks to `git annex`, the following command will copy the directory structure and some small files of your dataset on `data.neuro`:
+```
+cd data_nvme_<POLYGRAMES_USERNAME>/<PROJECT_NAME>
+git clone git@data.neuro.polymtl.ca:datasets/<PROJECT_DATASET> 
```

Comment:
@sandrinebedard gave me a tour of how she handled https://github.com/sct-pipeline/ukbiobank-spinalcord-csa last year.

In short, she wrote a detailed process guide in

https://github.com/sct-pipeline/ukbiobank-spinalcord-csa/blob/2f2cec3b91294153a635a8f725c0fbc749d172ca/README.md?plain=1#L42-L43

https://github.com/sct-pipeline/ukbiobank-spinalcord-csa/blob/2f2cec3b91294153a635a8f725c0fbc749d172ca/README.md?plain=1#L94-L100

https://github.com/sct-pipeline/ukbiobank-spinalcord-csa/blob/2f2cec3b91294153a635a8f725c0fbc749d172ca/README.md?plain=1#L111-L120

While explaining, she realized some parts were left undocumented: she set up a conda environment to avoid surprise breakage, which may cause different results since [requirements.txt doesn't specify versions](https://github.com/sct-pipeline/ukbiobank-spinalcord-csa/blob/2f2cec3b91294153a635a8f725c0fbc749d172ca/requirements.txt#L1-L8), and there is a `git tag` named "r20210928" attached to the precise dataset she analysed

```
p115628@joplin:~/datasets/uk-biobank-processed$ git remote -v
origin  git@data.neuro.polymtl.ca:datasets/uk-biobank-processed (fetch)
origin  git@data.neuro.polymtl.ca:datasets/uk-biobank-processed (push)
p115628@joplin:~/datasets/uk-biobank-processed$ git log HEAD~..
commit 7ed28cd0aacaab0ce1570bccdba3bff495b5f496 (HEAD -> master, tag: show, tag: r20210928, origin/master, origin/HEAD)
Author: Sandrine Bedard <sandrine.bedard@polymtl.ca>
Date:   Sun Sep 5 17:49:04 2021 -0400

    change suffix in derivatives from labels-manual to labels-disc-manual
```

**but** she hasn't documented that that's the tag to go with the project. Otherwise her procedure fits [option 3](https://github.com/neuropoly/data-management/issues/136#issuecomment-941575717): write down all the steps and expect people will read all your docs and follow all your instructions.

Maybe you can already tell but I'm leaning against option 3. I think we need something more automated because if there's one thing I know about human-computer interaction it's that people *don't* read instructions. I don't know if `git submodule` (option 1) is the right answer but I think we need something automated. And we will know if we have it when we can _run a weekly cronjob on each of our papers that **reproduces** the final paper including all figures and statistical analyses from just source code and access to the data server_.

Reply:
Oh perf, I just poked around a bit and noticed I guess that you were trying to reproduce this project: https://github.com/NadiaBlostein/Chemical_Structure_Reconstruction

The [very first thing](https://github.com/NadiaBlostein/Chemical_Structure_Reconstruction/blob/eb5e4b260de5ebc67e0492975574356f8a1c4527/README.md?plain=1#L29-L33) it says pretty much is:

> 1. Download the Mordred compound sets (1 through 3) and place them in a `data/` folder

I have no idea where to get the Mordred compound sets (1 through 3) and what version (if any) of them I'm supposed to get and I don't know if they assume the `data/` folder is supposed to be in the same folder as the code, or if they mean in the current working directory (read it in the unix convention, I would read that as the current working directory!) which isn't necessarily the same.

And 

> 3. Run the main function in `src/thresholding.py`. 

There is no main function in that file. As a programmer I can read the code and interpret that they must mean "run `python src/thesholding.py`", but if working with a large series of projects it would be impossible to track down and correct every small gap like that.

Their repo also depends on pandas and some other third-party python packages, but doesn't declare a setup.py or requirements.txt or a conda environment or anything. It doesn't even say what version of python it was written against. All of that is key to write down for reproducibility.

Those are some prime examples of the mistakes I want us to be able to avoid.